src/transport/HighPressureGasTransport.cpp

```diff
@@ -7,318 +7,511 @@
+
+    double mu_0 = 0.0;
+    for (size_t i = 0; i < 9; i++) {
+        mu_0 += c[i]*pow(T0,(i+1.0-4.0)/3.0);
     }
+    return correlation_viscosity_conversion*mu_0;
 }
 
-void HighPressureGasTransport::getMultiDiffCoeffs(const size_t ld, double* const d)
+double HighPressureGasTransport::elyHanleyReferenceThermalConductivity(double rho0,
+                                                                       double T0)
 {
-    // Not currently implemented.  m_Lmatrix inversion returns NaN.  Needs to be
-    //   fixed.  --SCD - 2-28-2014
-    throw NotImplementedError("HighPressureGasTransport:getMultiDiffCoeffs");
-    // Calculate the multi-component Stefan-Maxwell diffusion coefficients,
-    // based on the Takahashi-correlation-corrected binary diffusion coefficients.
+    // Computing the individual terms of Equation 18, ely-hanley1983 . This is an
+    // evaluation of the expressions shown in Table I of ely-hanley1983 . The
+    // correlation returns values of thermal conductivity in mW/m/K,
+    // so a conversion is needed.
+    const double correlation_factor = 1e-3; // mW/m/K to W/m/K
 
-    // update the mole fractions
-    update_C();
+    // This is the reference gas, dilute gas viscosity (eta_0 in Table III of
+    // ely-hanley1981)
+    double mu_0 = elyHanleyDiluteReferenceViscosity(T0);
 
-    // update the binary diffusion coefficients
-    update_T();
-    updateThermal_T();
+    // First term in Equation 18. This expression has the correct units because
+    // it does not use any empirical correlation, so it is excluded at the end from
+    // the unit conversion.
+    double Lambda_ref_star = (15*GasConstant / (4*ref_MW))*mu_0;
 
-    // Correct the binary diffusion coefficients for high-pressure effects; this
-    // is basically the same routine used in 'getBinaryDiffCoeffs,' above:
-    size_t nsp = m_thermo->nSpecies();
-    vector<double> molefracs(nsp);
-    m_thermo->getMoleFractions(&molefracs[0]);
-    update_T();
-    // Evaluate the binary diffusion coefficients from the polynomial fits -
-    // this should perhaps be preceded by a check for changes in T, P, or C.
-    updateDiff_T();
+    // Second term in Equation 18
+    const vector<double> b = {-2.52762920e-1, 3.34328590e-1, 1.12, 1.680e2};
+    double Lambda_ref_1 = (b[0] + b[1]*pow(b[2] - log(T0/b[3]), 2))*rho0;
+
+    // Third term in Equation 18
+    const vector<double> a = {-7.197708227, 8.5678222640e1, 1.2471834689e1,
+                              -9.8462522975e2, 3.5946850007e-1, 6.9798412538e1,
+                              -8.7288332851e2};
+    double delta_lambda_ref = exp(a[0] + a[1]/T0) * (exp((a[2] + a[3]*pow(T0,-1.5))*pow(rho0,0.1) + (rho0/ref_rhoc - 1)
+                *sqrt(rho0)*(a[4] + a[5]/T0 + a[6]*pow(T0,-2))) - 1.0);
 
+    return Lambda_ref_star + (Lambda_ref_1 + delta_lambda_ref)*correlation_factor;
+}
+
+
+void HighPressureGasTransport::getBinaryDiffCoeffs(const size_t ld, double* const d)
```

Comment:
I totally messed that one up. Thought that the other methods utilized the binary diffusion coefficient method to compute their mixture averaged quantities.

Reply:
I feel like duplicating the docstrings for the overridden methods is a bit redundant. Should I have a comment saying that the method is the same as the base classes's method, but with the m_bdiff quantity scaled by the Takahashi factor?

Reply:
I pushed some updates, and I've added those methods to the high pressure classes. I'm not satisfied with what I've done due to the amount of code duplication and the overall repeated operations done across all those diffusion coefficient methods, but I haven't figured out how to make it more efficient.

Reply:
It does feel like there's more duplicated code than necessary, but all the alternatives I can come up with have their own problems. I think the current solution is okay, and at least has the benefit of not increasing the complexity of the more frequently-used ideal gas transport implementation.

Reply:
Oh, I see. Really, there *should* be a common base class for `HighPressureGasTransport` and `ChungHighPressureGasTransport` other than just `MixTransport`. This would also let you deduplicate several other methods like `Tcrit_i` and friends, `initializeCriticalProperties`. I suppose the most straightforward name for it would be `HighPressureGasTransportBase`. In addition to the methods implemented in this class, you'd just need to add `m_P_corr_ij` as a virtual method.